core/gesture.js

```diff
@@ -651,11 +651,25 @@ Blockly.Gesture.prototype.handleWsStart = function(e, ws) {
   this.setStartWorkspace_(ws);
   this.mostRecentEvent_ = e;
   this.doStart(e);
+  if (!this.startBlock_ && !this.startBubble_) {
+    this.fireWorkspaceClick_(ws);
+  }
   if (this.startWorkspace_.keyboardAccessibilityMode) {
     Blockly.navigation.setState(Blockly.navigation.STATE_WS);
   }
 };
 
+/**
+ * Fires a workspace click event.
+ * @param {!Blockly.WorkspaceSvg} ws The workspace that a user clicks on.
+ * @private
+ */
+Blockly.Gesture.prototype.fireWorkspaceClick_ = function(ws) {
+  var clickEvent = new Blockly.Events.Ui(null, "workspaceClick", null, null);
```

Comment:
nit: Use `'` instead of `"` https://google.github.io/styleguide/jsguide.html#features-strings-use-single-quotes

Reply:
Updated in #3959. Just curious, should the eslint catch this?

Reply:
Yeah it totally could, although we use `"` in a lot of our test files, so might require a little refactoring at first.
https://eslint.org/docs/rules/quotes
From the goog-eslint-config: 
```js
    'quotes': ['error', 'single', {allowTemplateLiterals: true}],
```